Casks/p/publii.rb

```diff
@@ -1,18 +1,21 @@
 cask "publii" do
   arch arm: "apple-silicon", intel: "intel"
+  livecheck_arch = on_arch_conditional arm: "Apple Silicon", intel: "Intel Chip"
 
-  version "0.46.4"
-  sha256 arm:   "99dbb99716aaffec27c846f29f89260702e6a1c997e72dfe514be9b12698fb6d",
-         intel: "4058915775162e0ef0cddf7989c9facc7f12ea9b561ec7e809b64304663501d9"
+  version "0.46.4,17047"
+  sha256 :no_check # required as upstream package is updated in-place
 
-  url "https://getpublii.com/download/Publii-#{version}-#{arch}.dmg"
+  url "https://getpublii.com/download/Publii-#{version.csv.first}-#{arch}.dmg"
   name "Publii"
   desc "Static website generator"
   homepage "https://getpublii.com/"
 
   livecheck do
     url "https://getpublii.com/download/"
-    regex(/href=.*?Publii[._-]v?(\d+(?:\.\d+)+)(?:[._-]#{arch})?\.dmg/i)
+    regex(/macOS\s+#{livecheck_arch}.*?v?(\d+(?:\.\d+)+)\s+\(build\s+(\d+)\)/i)
+    strategy :page_match do |page, regex|
+      page.scan(regex).map { |match| "#{match[0]},#{match[1]}" }
```

Comment:
```suggestion
    regex(%r{\(build\s+(\d+)\).*?href=["']?[^"' >]*?Publii[._-]v?(\d+(?:\.\d+)+)[._-]#{arch}\.dmg}im)
    strategy :page_match do |page, regex|
      page.scan(regex).map { |match| "#{match[1]},#{match[0]}" }
```

I'm not sure if the approach in this suggestion is necessarily better but we could omit `livecheck_arch` if we match the filename instead.

Reply:
Thanks, it's definitely simpler.